### PR TITLE
Improve JS package descriptions

### PR DESCRIPTION
**Pull request by chrisvanpatten**
## Description
I went in to fix a typo in the `@wordpress/components` package description, and ended up giving them all a bit of love.

## How has this been tested?
It has not been tested — the changes are only to text in the packages' `package.json` files.

## Types of changes
+ Applied sentence case to all descriptions
+ Expanded abbreviations (to words like internationalization, accessibility, utilities, etc.)
+ Removed trailing periods on a few descriptions
+ Improved the language in one or two cases

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Nice one 💯 Thanks for opening this PR. In general, I like those changes because they bring consistency. I would prefer to have someone else to double check them before we proceed with mthe erge.

File: packages/babel-plugin-import-jsx-pragma/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "@wordpress/babel-plugin-import-jsx-pragma",
 	"version": "1.0.1",
-	"description": "Babel transform plugin for automatically injecting an import to be used as the pragma for the React JSX Transform plugin.",
```
Comment (tofumatt) [inline packages/babel-plugin-import-jsx-pragma/package.json]: Interesting, I'd have thought sentence-case would be better for those. Some descriptions are surely multi-sentence?

It's split on NPM (React seems to be all without periods; lodash all with, etc.)

Given we're in the "comments must have full-stops" camp (I like it), maybe we should be here too? 🤷‍♂️

File: packages/components/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "@wordpress/components",
 	"version": "1.0.1",
-	"description": "UI Compoonents for WordPress",
```
Comment (tofumatt) [inline packages/components/package.json]: 😅

File: packages/compose/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "@wordpress/compose",
 	"version": "1.0.1",
-	"description": "WordPress Higher Order components",
+	"description": "WordPress higher-order components",
```
Comment (tofumatt) [inline packages/compose/package.json]: I'd add "(HOC)" in brackets here; lots of people throw that acronym around and giving newbies context here for free is 👍

File: packages/keycodes/package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "@wordpress/keycodes",
 	"version": "1.0.1",
-	"description": "KeyCodes utilities for WordPress",
+	"description": "Keycodes utilities for WordPress",
```
Comment (tofumatt) [inline packages/keycodes/package.json]: This one, I think, could be expanded upon further ("Keycodes utilities for WordPress; used to check for keyboard events across browsers/operating systems")

**Review by tofumatt — APPROVED** [review]
I dig it. A few thoughts but this is better than before so 🚢. If you want to make any of the small changes I suggest then feel free to do so before merge 😄

**Comment by gziolo** [comment]
Travis failed after I added fixes suggested by @tofumatt. Those are only text changes so I will merge as is.

File: packages/babel-plugin-import-jsx-pragma/package.json (hunk 1 above)
Comment (gziolo) [reply packages/babel-plugin-import-jsx-pragma/package.json]: > Given we're in the "comments must have full-stops" camp (I like it), maybe we should be here too?

I would be also happy to see this change.

Comment (aduth) [reply packages/babel-plugin-import-jsx-pragma/package.json]: Seems like something that would be good to have (though from what I can tell, not yet offered as a feature) from [`npm-package-json-lint`](https://github.com/tclindner/npm-package-json-lint), which we're already using and [have a predefined configuration](https://github.com/WordPress/gutenberg/tree/master/packages/npm-package-json-lint-config) for.

Comment (tclindner) [reply packages/babel-plugin-import-jsx-pragma/package.json]: Hey! I just released v3.3.0 of [`npm-package-json-lint`](
https://github.com/tclindner/npm-package-json-lint) with a new [`description-format`](https://github.com/tclindner/npm-package-json-lint/wiki/description-format) rule. It has two options:

1. `requireCapitalFirstLetter` - Throws an error if the first character in the description isn't capitalized.
2. `requireEndingPeriod` - Throws an error if the description doesn't end with a period.

Hopefully this works well for your team!

Comment (gziolo) [reply packages/babel-plugin-import-jsx-pragma/package.json]: Thanks, opened PR to start using this new rule: https://github.com/WordPress/gutenberg/pull/8596.

**Pull request merged**